### DISCUSSION TITLE
Addition of multiple extruders

### DISCUSSION THREAD

**Discussion by terryrobison** [OrcaSlicer/OrcaSlicer#931] [Ideas]
It would be fantastic to have multiple extruders added to OrcaSlicer.  My printer has four extruders, two with 0.4 mm nozzles, another with a 0.6mm, and another with a 0.8mm.   Typically different materials are loaded in each tool.  Both Slic3r and Superslicer can add multiple extruders and profiles.  Can this functionality be added to OrcaSlicer?  It would be nice to perform all my slicing needs with one slicer.  
Regards,

**Comment by freeformz** [comment]
+1 as now the Prusa XL is shipping with up to 5 toolheads

**Comment by terryrobison** [comment]
Awesome, thanks for the update!

On Tue, 26 Sept 2023 at 02:16, Edward Muller ***@***.***>
wrote:

> +1 as now the Prusa XL is shipping with up to 5 toolheads
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SoftFever/OrcaSlicer/discussions/931#discussioncomment-7103828>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIV76NAWPEANA4AFPSGAF4TX4GU65ANCNFSM6AAAAAAXT6K264>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Terry

"You can log off / But you can never leave."
 --------------------------------------------------------------
 -  Terry Robison
 -  Senior Systems Analyst
 -  Blueprint I.T. Services Pty. Ltd.
 -  ***@***.***         _\|/_
 -                                                   (o o)
 -------------------------------------oOO-(_)-OOo----

**Comment by Luigi084** [comment]
I'm looking too for dual nozzle for my Klipper modified Raise3D N2 plus. I'm still under prusaslicer for dual nozzle and orca for single noozzle.

**Comment by cdsmakestuff** [comment]
More / better support for multiple extruders would be great! Would love to use my IDEX more often.

**Comment by lukasschwoerer** [comment]
This would be great!!! I would rly like to switch to Orca Slicer for my E3D Toolchanger. But this is simply not possible at the moment =(

**Reply by Astrocat505** [reply]
What slicer are you using currently for your E3d Tool changer? I have a ProForge 4 and it's based on the E3d.

**Reply by lukasschwoerer** [reply]
Currently using PrusaSlicer which is working reasonably well... I can send you the profile if you want.
Which slicer do you use?

**Comment by CornBred89** [comment]
I want to 100% switch from Prusa Slicer to Orca Slicer for my Sovol SV04. However, Orca Slicer does not support multi extruder printers apparently. So now I'm stuck using 2 different slicers. Prusa Slicer for dual material and Ocra for single extruder prints. Quite bummed :(

**Reply by CornBred89** [reply]
I don't understand how they have profiles for the RatRig V-Cast and the Snapmaker J1. Both are IDEX machines. Are they really limited to using just one extruder in Orca Slicer? More info would be great.

**Reply by bn1** [reply]
J1 owner here: actually yes, I'm using Prusa Slicer for IDEX prints and Orca for the rest. I was not able to figure out how to use Orca for IDEX reasonably well. (eg. it doesn't pre-heat other head, can't use wipe tower efficiently etc.)

**Comment by PLMD86** [comment]
I also have a printer with 4 extruders and I would like to use Orca....

**Comment by Hellsparks** [comment]
Im working on a toolchanger for my Voron 2.4, having multiple toolhead support would be a GODSEND as i realy prefer orca..

**Reply by PLMD86** [reply]
In theory you can use more extruders (I currently use them with Orca) but with Prusa Slicer it is easier to configure them... in the sense that you can set the diameter of the nozzle and everything else much faster

**Reply by BlackShadeOSS** [reply]
How can you do that?

**Comment by sick4noise** [comment]
I just built a 6-in-1 printer and I would love to be able to use Orca. I hate having to use SuperSlicer just for multi-color prints - I find the UI much more difficult to use.

**Comment by terryrobison** [comment]
Try this:
https://github.com/supermerill/SuperSlicer/releases/tag/2.5.59.6


On Thu, 18 Jan 2024 at 02:36, R Derek Burket ***@***.***>
wrote:

> I just built a 6-in-1 printer and I would love to be able to use Orca. I
> hate having to use SuperSlicer just for multi-color prints - I find the UI
> much more difficult to use.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SoftFever/OrcaSlicer/discussions/931#discussioncomment-8158401>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIV76NHITAZ2LKAUEDTH24TYO7VWBAVCNFSM6AAAAAAXT6K266VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DCNJYGQYDC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Terry

"You can log off / But you can never leave."
 --------------------------------------------------------------
 -  Terry Robison
 -  Senior Systems Analyst
 -  Blueprint I.T. Services Pty. Ltd.
 -  ***@***.***         _\|/_
 -                                                   (o o)
 -------------------------------------oOO-(_)-OOo----

**Reply by cryoglobe** [reply]
last version I used of SuSi was having bugs ... and now lots of new features of Prusa 2.7 are still not included :(   
Honestly , with Orca on the scene ... I have some difficulties to see what SuSi will bring as added value

**Comment by BlackShadeOSS** [comment]
any updates on this?

**Comment by BrewNinja** [comment]
Add another vote for this. Having a toolchanger and like the interface of Orca so much better, I would love to see it added!

**Comment by traxman25** [comment]
Another one with 6 tool changer here also. Would really prefer to stick with Orca Slicer as it does such a great job.

**Comment by akinferno** [comment]
Just adding my voice to the mix. Multi-tool support is baked into PrusaSlicer and Cura. Would love to just use Orca, instead.

**Comment by Andy-1664** [comment]
+1 here too, defiantly a feature that will become more mainstream

**Comment by akinferno** [comment]
I didn't provide much incentive, besides adding my voice, but there here are some of the reasons it makes sense:
- There are several commercially available IDEX printers: Sovol S04 (<$500), Prusa XL, Snapmaker J1S, Raise3D E2CF, Flashforge Creator 3 Pro IDEX and many others.
- Voron Community working toolchanger projects: Tapchanger, Stealthchanger, Daksh, with more in development. Mihai Designs PitStop. IDEX versions of the Voron 0, Trident and Voron 2. 
- Lots of older community projects that have been around a while, on Thingiverse and Printables. The community is growing due to the popularity of filament changing systems, which produce a lot of waste due to purges.

**Comment by phongshader** [comment]
+1

**Comment by DrKlipper** [comment]
I get my first 2 in 1 printer this week. 
It would be awesome if Orca can support multi extruder like Prusa. 
Thx

**Comment by PauloCabrita** [comment]
One more for the multiple extruders support.

**Comment by Raymond3D** [comment]
+1. Would really be great to have

**Comment by erazor666** [comment]
Stealthchanger had serialized 12 machines last I looked with a lot more in the works, myself included. Thumbs up for this feature to be added to my favorite and only slicer really!

**Reply by BlackShadeOSS** [reply]
13 not 12 :)

**Comment by Naam** [comment]
+1 would be nice to port the feature from SuperSlicer:
> Multi-material (multiple extruders) object printing
> Standby temperature and automatic wiping for multi-extruder printing

I own a CreatBot F430, which has two extruders 73mm apart and can help with testing if needed.
I am not satisfied with CreatBot's slicer, which is a poor modification of Slic3r. It looks like Prusa Slicer also supports multiple extruders, but CreatBot's slicer has some secret sauce that makes it work better than Prusa out of the box for the printer. If possible, I'd really like to work with Orca.

**Comment by Festivejelly** [comment]
I use an e3d tool changer and OrcaSlicer is preferable to me. I get around it by adding multiple filaments and using gcode at the start and end of the filament to swap tools. But It lacks some basic features for multi extruders, such as being able to choose when is extruder a feature is printed with. For example if I wanted T0 to print the outer wall but T1 to print inner walls. Same for skin, being able to use a different extruder to print the top or bottom would be handy too.

I know its quite a niche request to be able to do these things but it sure would make my workflow easier!

**Reply by Astrocat505** [reply]
Which slicer are you using for the E3d Tool changer currently? Cura is the only slicer with a profile for my ProForge 4 and I'd like to change that.

**Comment by ariekraakjr** [comment]
+1 from me. Just finished my Double Dragon and have to use Prusa Slicer for that printer. (don´t like)

**Comment by broncosis** [comment]
+1 here too  I just finished building my voron tap changer

**Comment by adbizau** [comment]
+1 please. Voron tapchanger

**Comment by SlinkySlonkyOlivia** [comment]
+1 from me, cant use my ultimaker 2 printhead in orcaslicer, or future builds of mine that will be toolchangers

**Comment by Astrocat505** [comment]
Bumping this idea. I'd like to use Orca slicer for my ProForge 4 tool changer like I do with my other 2 printers.

**Comment by SimonSolar2C** [comment]
Absolutely love this idea - please I wan to use OrcaSlicer on my Creatbot f430 dual extruder

**Comment by AstronIsTaken** [comment]
Please add multi extruder support. Looking forward to use Orca with my Voron toolchanger

**Comment by webboty** [comment]
+1 vote from my end. Orca Slicer is one of the best slicers on the market but unfortunately lacking the option of multiple extruders which makes it fall way behind any other slicer that does support it right now. Time to pick up.

**Comment by hliebscher** [comment]
I have the prusa XL 5 T running works great
![prusa xl](https://github.com/SoftFever/OrcaSlicer/assets/9274162/0cc20642-5199-4d93-8fbf-f12319f687bb)

**Reply by Filpereira** [reply]
Hi, can you explain how you did this, I would like to try it on my xl 5tools

**Comment by ariekraakjr** [comment]
That´s multi filaments, not multi toolheads.

**Comment by radosaga** [comment]
Theoretically you can put your custom Gcode for filament start/end in the advanced filament setting. But there will be some downsides of course

![custom_filament_G-code](https://github.com/SoftFever/OrcaSlicer/assets/132309748/2b303899-64b9-4270-a352-d08192e99a5e)

**Reply by ariekraakjr** [reply]
It's a workaround, not a solution.

**Reply by radosaga** [reply]
yes I fully agree, but after certain time a workaround can be considered as a solution :)

**Reply by BrewNinja** [reply]
Not really. How are you handling different nozzle sizes? Oh wait....

**Reply by radosaga** [reply]
Not tested yet since my voron is still a single-toolhead, therefore I started with "theoretically" :)

**Reply by BrewNinja** [reply]
My point was multi-filament works now, but mulit-tool or extruder is not possible.

**Comment by diode-ee** [comment]
Another vote to make this happen. I'm getting ready to build Draftshift Stealthchanger and would really like to have this as a feature of the slicer I've been using.

**Comment by DrKlipper** [comment]
So many answers and votes in this topic. But none of the devs gives a note if this request is possible or not. 
Maybe there should be a linked issue to this topic to get a little bit more attention?
Or at least get it into the focus of the devs ...

**Reply by webboty** [reply]
You are right about this. This is also the reason why I gave up on Orca Slicer and went back to prusa slicer because of the missing feature of multiple extruders.

**Comment by BrewNinja** [comment]
I did open an issue for it a few days ago because I didn't know if one had been created and linked. You can find it here: https://github.com/SoftFever/OrcaSlicer/issues/5079

**Reply by DrKlipper** [reply]
Maybe you should add a link to this discussion ...

**Reply by BrewNinja** [reply]
Good call! Thought I had done that already.

**Comment by DrKlipper** [comment]
Maybe we need also some Feedback in the issue. 
I will make a start :-)

**Comment by StealthViper27** [comment]
Adding my voice to the choir. Please add this to Orca!

**Comment by BrewNinja** [comment]
If everyone here could also go add a thumbs up to the issue and add a comment there as well, it would be helpful!

This is the link: https://github.com/SoftFever/OrcaSlicer/issues/5079

**Reply by Astrocat505** [reply]
Done

**Reply by diode-ee** [reply]
Added me comment as well.

**Comment by freeformz** [comment]
@SoftFever What can we do to help with this? Do you have an IDEX printer? Maybe we can see about raising some funds for one? (I already sponsor, but that is just in general). Maybe someone from Prusa could donate (at least temporarily) a multi tool head XL? Maybe the folks behind one of the other IDEX printers could donate (at least temporarily) one of their IDEX printers? Something else? While I am a "programmer" this whole area and code base are not in my wheel house, so I don't feel comfortable attempting to help more actively.

**Reply by freeformz** [reply]
But I am willing to test since I do have a 5 TH Prusa XL.

**Comment by broncosis** [comment]
I would put a little money on this as well if that would help I am sure there's a few of us that could put up a few bucks. We could easily cover a idex.

⁣Get BlueMail for Android ​

**Comment by SlinkySlonkyOlivia** [comment]
Im willing to help out using my own printers, i have a dual nozzle modded ultimaker original running klipper

**Comment by SoftFever** [comment]
Thanks all for the feedback.
I will take a look

**Comment by broncosis** [comment]
@softfever. Thanks for taking the time to look there is a few of us and a growing number of tool changers coming soon. 

⁣Get BlueMail for Android ​

**Comment by cjackson234** [comment]
Happy to test when its ready

**Comment by akinferno** [comment]
I am a willing test multi-extruder support, as well.

**Comment by lukasschwoerer** [comment]
@SoftFever I have a 4 TH E3D Tool changer on which I happily run tests if you need it.

**Comment by Simulacra101** [comment]
First of all I wanted to thank you for making such an amazing slicer. For single material printing I couldn't have a better slicer to cover all my printers than orca. Even multicolor works great with it in some regards. Sadly multi material prints are next to impossible with only having one configurable process ore one single tool head. Personally I have a batch of fully customized idex printers running Klipper that have in itself two completly different tool heads and nozzle sizes. Having the option to not just customly change the tool heads via custom gcode, but rather change the whole slicing process per tool head would be a great way too fully use the potential of not just those but rather every multi tool/extruder headed printers out there. Since multi tool headed printers are getting more and more popular it would be a great implementation

**Comment by DrewCPU** [comment]
Happy to test with my Sovol SV04 IDEX printer!

**Comment by adrianbear** [comment]
+1 from ABear, Orca is my slicer of choice and I too have a tool changer :)

**Comment by freeformz** [comment]
https://github.com/SoftFever/OrcaSlicer/pull/6087

**Comment by Emperornero** [comment]
Is this being worked on for IDEX machines?   Have a RR V4 and honestly hate Prusa Slicer, I use Orca for all my other printers.  Would prefer to stick to Orca for all printers.

**Comment by CornBred89** [comment]
There's a beta version on the github that supports multitool machines.
